modin/experimental/xgboost/xgboost.py

```diff
@@ -0,0 +1,499 @@
+                actors, actors_ips, row_partitions
+            )
+
+        if data_for_aligning is None:
+            for actor_ip in actors_ips:
+                actor_row_partitions = []
+                order_of_indexes = []
+                for i, row in enumerate(row_partitions):
+                    if partitions_ips[i] in actor_ip:
+                        actor_row_partitions.append(row[1])
+                        order_of_indexes.append(i)
+
+                row_partitions_by_actors.append(
+                    (actor_ip, order_of_indexes, actor_row_partitions)
+                )
+        else:
+            for (actor_ip, order_of_indexes, _) in data_for_aligning:
+                actor_row_partitions = []
+                for row_idx in order_of_indexes:
+                    actor_row_partitions.append(row_partitions[row_idx][1])
+
+                row_partitions_by_actors.append(
+                    (actor_ip, order_of_indexes, actor_row_partitions)
+                )
+
+    return row_partitions_by_actors
+
+
+def _get_evenly_distr_of_partitions_by_ips(actors, actors_ips, row_partitions):
+    num_actors = len(actors)
+    row_parts_last_idx = (
+        len(row_partitions) // num_actors
+        if len(row_partitions) % num_actors == 0
+        else len(row_partitions) // num_actors + 1
+    )
+    row_partitions_by_actors = []
+    start_idx = 0
+    for actor, ip in zip(actors, actors_ips):
+        first = start_idx
+        last = (
+            (start_idx + row_parts_last_idx)
+            if (start_idx + row_parts_last_idx < len(row_partitions))
+            else len(row_partitions)
+        )
+
+        actor_row_partitions = []
+        order_of_indexes = []
+        while first < last:
+            actor_row_partitions.append(row_partitions[first][1])
+            order_of_indexes.append(first)
+            first += 1
+        start_idx += row_parts_last_idx
+
+        row_partitions_by_actors.append((ip, order_of_indexes, actor_row_partitions))
+
+    return row_partitions_by_actors
+
+
+def train(
+    params: Dict,
+    dtrain: ModinDMatrix,
+    *args,
+    evals=(),
+    nthread: Optional[int] = cpu_count(),
+    evenly_data_distribution: Optional[bool] = True,
+    **kwargs,
+):
+    """
+    Train XGBoost model.
+
+    Parameters
+    ----------
+    params : dict
+        Booster params.
+    dtrain : ModinDMatrix
+        Data to be trained.
+    nthread: int
+        Number of threads for using in each node. By default it is equal to
+        number of threads on master node.
+    evenly_data_distribution: boolean, default True
+        Whether make evenly distribution of partitions between nodes or not.
+        In case `False` minimal datatransfer between nodes will be provided
+        but the data may not be evenly distributed.
+    \\*\\*kwargs:
+        Other parameters are the same as `xgboost.train` except for
+        `evals_result`, which is returned as part of function return value
```

Comment:
why diverge from xgboost API?

Reply:
This is made accordance with [xgboost.dask](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/dask.py#L871) API for aligning with distributed API.